cmd/snap/cmd_quota.go

```diff
@@ -316,12 +444,47 @@ func (x *cmdQuotas) Execute(args []string) (err error) {
 			return fmt.Errorf("internal error: constraints is missing from daemon response")
 		}
 
-		constraintVal := "memory=" + strings.TrimSpace(fmtSize(int64(q.Constraints.Memory)))
-		currentVal := ""
-		if q.Current != nil && q.Current.Memory != 0 {
-			currentVal = "memory=" + strings.TrimSpace(fmtSize(int64(q.Current.Memory)))
+		var grpConstraints []string
+
+		// format memory constraint as memory=N
+		if q.Constraints.Memory != 0 {
+			grpConstraints = append(grpConstraints, "memory="+strings.TrimSpace(fmtSize(int64(q.Constraints.Memory))))
+		}
+
+		// format cpu constraint as cpu=NxM%,cpu-set=x,y,z
+		if q.Constraints.CPU != nil {
+			if q.Constraints.CPU.Count != 0 || q.Constraints.CPU.Percentage != 0 {
```

Comment:
This condition seems to be useless, as you already have the right checks in the inner block.